### ISSUE TITLE
Storybook addon-a11y fails with a11y errors about aria-label

### ISSUE THREAD

**Issue by stufreen** [Shopify/polaris-react#4447]
## Issue summary

New versions of `axe-core` (4.2 and later) will treat it as a violation if you have `aria-label` attribute on generic elements like `span`. See the Axe [changelog](https://github.com/dequelabs/axe-core/blob/develop/CHANGELOG.md#bug-fixes-7) and this [PR](https://github.com/dequelabs/axe-core/issues/2766). Polaris does this on the `Icon` component and maybe others as well.

This is significant because recent versions of `@storybook/addon-a11y` use axe-core 4.2.0. See @kaelig PR [here](https://github.com/storybookjs/storybook/pull/14749).

Projects using Polaris and `addon-a11y` in their CI pipeline will either have to disable this rule or cannot upgrade Storybook. 

## Expected behavior

The a11y rule linked in the [axe-core PR](https://github.com/dequelabs/axe-core/issues/2766) seems valid. Screen readers apparently won't read `aria-label` for generic elements, so they should be removed.

## Actual behavior

Polaris includes `aria-label` for many generic elements.

## Steps to reproduce the problem

1. Install @storybook/addon-a11y version 6.3.7 and Polaris on the same project
2. Run storybook
3. Open accessibility tab for a story that includes one of the affected components (e.g. Icon with `accessibilityLabel` prop)
4. A violation will appear relating to the `aria-label` attribute

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 6.5.0
- Browser: NA
- Device: NA
- Operating System: NA
